fpga_interchange/populate_chip_info.py

```diff
@@ -1740,6 +1760,426 @@ def get_cell(cell_type):
             chip_info.macros.append(macro)
 
 
+def clusters_from_macros(device, phy_placements, debug=False):
```

Comment:
this requires a docstring explaining what the function does and briefly also the algorithms used